### PR TITLE
feat(api): add endpoint to get contact templates

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

In order to prepare auto imports features, we need to implement an endpoint to get the list of contact templates.

**Fixes** # MON-12595

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: config/routes/Centreon/configuration/contact_template.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+configuration_contact_template_find_all:
+  methods: GET
+  path: /configuration/contact/template
```
Comment (a-launois) [inline config/routes/Centreon/configuration/contact_template.yaml:3]:
```suggestion
  path: /configuration/contacts/templates
```

Comment (jeremyjaouen) [reply config/routes/Centreon/configuration/contact_template.yaml:3]: I think we should discuss about this, as in my opinion a GET on contact/template without any id or parameters is plural as we need to get All, if i have /contact/template/{id} I need one. The endpoint should reflect the resource type but i'm totally open to discuss about it !

File: src/Core/Contact/Infrastructure/Repository/DbReadContactTemplateRepository.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+class DbReadContactTemplateRepository extends AbstractRepositoryDRB implements ReadContactTemplateRepositoryInterface
+{
+    /**
+     * @var SqlRequestParametersTranslator
+     */
+    private SqlRequestParametersTranslator $sqlRequestTranslator;
+
+    /**
+     * @param DatabaseConnection $db
```
Comment (kduret) [inline src/Core/Contact/Infrastructure/Repository/DbReadContactTemplateRepository.php:40]:
```suggestion
     * @param DatabaseConnection $db
     * @param SqlRequestParametersTranslator $sqlRequestTranslator
```

File: tests/php/Core/Contact/Application/UseCase/FindContactTemplatesTest.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\Contact\Application\UseCase;
```
Comment (kduret) [inline tests/php/Core/Contact/Application/UseCase/FindContactTemplatesTest.php:24]:
```suggestion
namespace Tests\Core\Contact\Application\UseCase;
```

**Review by kduret — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Pull request merged**